### PR TITLE
The static method java.lang.ThreadLocal.withInitial is added  On line 99

### PR DESCRIPTION
**Pull request by gaurangkudale**
Added method 

 public static <S> ThreadLocal<S> withInitial(Supplier<? extends S> supplier) {
    return new SuppliedThreadLocal<>(supplier);
  }

### CODE REVIEW

**Comment by cyrille-artho** [comment]
Thank you for your contribution. Is there a unit test available that fails without the new method, so we can confirm that it works as intended? If there is no existing test, could you please try to create one?

**Comment by gaurangkudale** [comment]
> Thank you for your contribution. Is there a unit test available that fails without the new method, so we can confirm that it works as intended? If there is no existing test, could you please try to create one?

@cyrille-artho Can you please help me with the unit test because I'm not familiar with it

**Comment by cyrille-artho** [comment]
The key challenge is to find a small program that fails without your patch but which works with it. (Possibly this could mean that under a certain version of the JDK, the code may not compile without the patch.)
After that, the program can be adapted as a unit test.
See https://github.com/javapathfinder/jpf-core/wiki/Writing-JPF-tests in the wiki for guidance on how to write unit tests.
I can help you with the final step, but as I said, finding an example is the most important part.

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho Can you please tell me how to find an example?

**Comment by cyrille-artho** [comment]
Hi Guarang,
Coming up with a good example/test can indeed be difficult. The best way to find code snippets/examples is by using a web search.
For example, I found these fragments that may make a working example (adding the assertion at the end):
```
ThreadLocal<Integer> threadLocal = ThreadLocal.withInitial(() -> 1);
Integer result = threadLocalValue.get();
assertEquals(result, 1);
```
I assume that this either won't compile without your patch (as constructor with the "supplier" lambda expression argument is not available without it) or will not return the correct value.
Try running a small program like this with the old JPF (without your patch) and the new JPF (with your patch) first, and then convert the program into a unit test using the instructions on the wiki (link in the above post).

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho,
I have written some code to test please look at it
` import javax.sound.sampled.SourceDataLine;

public class ThreadLocalInitialValueExample {
    public static <MyObject> void main(String[] args) {
        ThreadLocal<MyObject> threadLocal1 = new ThreadLocal<>(){
            @Override
            protected MyObject initialValue() {
                return new MyObject();
            }
        };
        ThreadLocal<MyObject> threadLocal2 = 
            ThreadLocal.withInitial(() -> {
                return new MyObject();
            });

        Thread thread1 = new Thread(() -> {
            System.out.println("ThreadLocal1: "+ threadLocal1.get());
            System.out.println("ThreadLocal2: "+ threadLocal2.get());
        });

        Thread thread2 = new Thread(() -> {
            System.out.println("threadLocal1: " + threadLocal1.get());
            System.out.println("ThreadLocal2: "+ threadLocal2.get());
        });

    }
}`

**Comment by cyrille-artho** [comment]
Hi,
Something seems to have gone slightly wrong when copying the code. Please copy complete code or link to an example. (Unfortunately, the simple issue tracker on GitHub does not support file attachments.)

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho ,
I've attached the link of file please g through it

https://kjeieduin-my.sharepoint.com/:t:/g/personal/14seit_tcoer_kjei_edu_in/EWkMXzPF1JFMkREt2iyXFWEBv68DWicZMbP0I1LnNRqyPw?e=PqsSmH

**Comment by gaurangkudale** [comment]
@cyrille-artho 
I've also created one more example for unit test and linked is attached below

https://kjeieduin-my.sharepoint.com/:t:/g/personal/14seit_tcoer_kjei_edu_in/EWSO7PLBETRKvm900jwuIHoB4bL0mHgCKyRrSPQ8pCKlew?e=3TkVKc

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
Did you check the unit test?

**Comment by cyrille-artho** [comment]
Could you please complete the unit test so it has an assertion that checks the behavior? The output is not easy to inspect when run on Gradle's test runner. The test code was also not 100 % complete so it does not compile without the additional imports for JUnit.

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho,
I have gone through the unit test but I'm not familiar with it can somebody help me to create a unit test?

**Comment by cyrille-artho** [comment]
As a first step, simply add one of your examples to the pull request (as a program that runs from `main`). I can then modify it to work as a unit test.

**Comment by gaurangkudale** [comment]
> As a first step, simply add one of your examples to the pull request (as a program that runs from `main`). I can then modify it to work as a unit test.

Hi @cyrille-artho, 
as you said I've added the example with the main method

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
did you checked the example?

**Comment by cyrille-artho** [comment]
Looks good, and I have triggered the CI build. I will run the tests on my local machine later this week or next week, as soon as I have time.

**Comment by gaurangkudale** [comment]
@cyrille-artho 
Sure

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho,
Added the new test file under  "src/tests"

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
Did you check the example is it good?

File: src/classes/java/lang/ThreadLocal.java
Comment (cyrille-artho) [inline src/classes/java/lang/ThreadLocal.java:24]: Remove these two unused imports.

File: src/tests/ThreadLocalTest/App.java
Comment (cyrille-artho) [inline src/tests/ThreadLocalTest/App.java:21]: Add a newline character (line break) at the end of the last line.

**Review by cyrille-artho — CHANGES_REQUESTED** [review]
Hi Guarang,
Great job! This version works perfectly. I'd just like you to make two small cosmetic changes before merging.

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho Changes are done

**Comment by cyrille-artho** [comment]
Hi,
There was a misunderstanding: What I meant was to remove the unused import statements in the implementation, and add a newline character at the end of the last source code line. You have added a print statement that does not compile. The previous revision was good except for these two minor issues (and for some reason, the PR did not show the completely refreshed version on GitHub, but I will look at this again later once you have a revised version that compiles and passes the tests).

**Comment by gaurangkudale** [comment]
> Hi, There was a misunderstanding: What I meant was to remove the unused import statements in the implementation, and add a newline character at the end of the last source code line. You have added a print statement that does not compile. The previous revision was good except for these two minor issues (and for some reason, the PR did not show the completely refreshed version on GitHub, but I will look at this again later once you have a revised version that compiles and passes the tests).

okay so now I only need to add a new line character like `\n` is that right

**Comment by gaurangkudale** [comment]
waiting for reply

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho 
Looking for your guidance regarding the Tests

**Comment by gaurangkudale** [comment]
In my local machine if I remove these two imports then it is showing me errors because I've used those packages in the unit test
I've attached the screenshot in the link 
maybe that's why it is not passing the test cases please look into it

[https://kjeieduin-my.sharepoint.com/:i:/g/personal/14seit_tcoer_kjei_edu_in/EemLvc6Z8oJChLmN0Csq6pkB-3Uqv2tkNfQpDquWWLerjg?e=MPvrXt](url)

File: src/classes/java/lang/ThreadLocal.java
Comment (gaurangkudale) [reply src/classes/java/lang/ThreadLocal.java]: Changes are done

**Comment by gaurangkudale** [comment]
Hi @cyrille-artho, 

- As you suggested the changes I've done the first change which is to remove unused imports from 'src/classes/java/lang/ThreadLocal.java'
- Can you please explain to me the second change so that I'll work on it

**Review by cyrille-artho — APPROVED** [review]

**Comment by cyrille-artho** [comment]
Looks good. Thank you very much for your effort and patience.

**Comment by gaurangkudale** [comment]
@cyrille-artho Thank you for guiding me for my First contribution really excited to do more

**Comment by cyrille-artho** [comment]
You're welcome. There are many open issues on the issue tracker left, and other features that are not yet tested (or documented as issues), for example, the `java.util.stream` API.

**Comment by gaurangkudale** [comment]
> You're welcome. There are many open issues on the issue tracker left, and other features that are not yet tested (or documented as issues), for example, the `java.util.stream` API.

Hi @cyrille-artho, 
Can you please share the more details about this issue so that I will work on it

**Comment by cyrille-artho** [comment]
I have not documented anything yet (and doing so on the issue tracker would be a good start).
The problem is the following:
We don't have any unit tests for the new Java streams collections (`java.util.stream`):

https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html

Simply including a few examples would be helpful to see if JPF can handle these cases at all (there may be missing native peer classes/methods), and what problems there might be.

I have created a new issue for this: https://github.com/javapathfinder/jpf-core/issues/302
So let's continue discussions there.

**Pull request merged**
